pkg/polaris/polaris_test.go

```diff
@@ -397,6 +398,225 @@ func TestAzureSubscriptionAddAndRemove(t *testing.T) {
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Add default Azure subscription to Polaris.
+	subscription := azure.Subscription(testSubscription.SubscriptionID, testSubscription.TenantDomain)
+	id, err := client.Azure().AddSubscription(
+		ctx,
+		subscription,
+		core.FeatureCloudNativeArchivalEncryption,
+		azure.Regions("eastus2"),
+		azure.Name(testSubscription.SubscriptionName),
+		azure.FeatureSpecificInfo(
+			&azuregql.FeatureSpecificInfo{
+				UserAssignedManagedIdentity: &azuregql.UserAssignedManagedIdentity{
+					Name:              testSubscription.Archival.ManagedIdentityName,
+					ResourceGroupName: testSubscription.Archival.ResourceGroupName,
+					PrincipalID:       testSubscription.Archival.PrincipalID,
+					Region:            azuregql.RegionEastUS2,
+				},
+			},
+		),
+		azure.ResourceGroup(
+			&azuregql.ResourceGroup{
+				Name:   testSubscription.Archival.ResourceGroupName,
+				Region: azuregql.RegionEastUS2,
+			},
+		),
+	)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify that the subscription was successfully added.
+	account, err := client.Azure().Subscription(
+		ctx,
+		azure.CloudAccountID(id),
+		core.FeatureCloudNativeArchivalEncryption,
+	)
+	if err != nil {
+		t.Error(err)
+	}
+	if account.Name != testSubscription.SubscriptionName {
+		t.Errorf("invalid name: %v", account.Name)
+	}
+	if account.NativeID != testSubscription.SubscriptionID {
+		t.Errorf("invalid native id: %v", account.NativeID)
+	}
+	if account.TenantDomain != testSubscription.TenantDomain {
+		t.Errorf("invalid tenant domain: %v", account.TenantDomain)
+	}
+	if n := len(account.Features); n == 1 {
+		if name := account.Features[0].Name; name != "CLOUD_NATIVE_ARCHIVAL_ENCRYPTION" {
+			t.Errorf("invalid feature name: %v", name)
+		}
+		if regions := account.Features[0].Regions; !reflect.DeepEqual(regions, []string{"eastus2"}) {
+			t.Errorf("invalid feature regions: %v", regions)
+		}
+		if account.Features[0].Status != "CONNECTED" {
+			t.Errorf("invalid feature status: %v", account.Features[0].Status)
+		}
+	} else {
+		t.Errorf("invalid number of features: %v", n)
+	}
+
+	// Update and verify regions for Azure account.
+	err = client.Azure().UpdateSubscription(ctx, azure.ID(subscription), core.FeatureCloudNativeArchivalEncryption,
+		azure.Regions("westus2"))
+	if err != nil {
+		t.Error(err)
+	}
+	account, err = client.Azure().Subscription(ctx, azure.ID(subscription), core.FeatureCloudNativeArchivalEncryption)
+	if err != nil {
+		t.Error(err)
+	}
+	if n := len(account.Features); n == 1 {
+		if regions := account.Features[0].Regions; !reflect.DeepEqual(regions, []string{"westus2"}) {
+			t.Errorf("invalid feature regions: %v", regions)
+		}
+	} else {
+		t.Errorf("invalid number of features: %v", n)
+	}
+	// Not removing storage account as there is support
+	// to remove non-native protection features in SDK.
+	// Will add removal logic once support is added.
```

Comment:
What is missing? Is it something in the SDK, in Polaris or a bit in both?

Reply:
There was a small bug in the test that I added earlier, now it is fixed.
Have added deletion of feature as well now.